packages/app-store/salesforce/lib/CrmService.ts

```diff
@@ -323,108 +323,190 @@ export default class SalesforceCRMService implements CRM {
     includeOwner?: boolean;
     forRoundRobinSkip?: boolean;
   }) {
-    const conn = await this.conn;
-    const emailArray = Array.isArray(emails) ? emails : [emails];
-    const appOptions = this.getAppOptions();
-    const recordToSearch =
-      (forRoundRobinSkip ? appOptions?.roundRobinSkipCheckRecordOn : appOptions?.createEventOn) ??
-      SalesforceRecordEnum.CONTACT;
-    let soql: string;
-    let accountOwnerId = "";
-    if (recordToSearch === SalesforceRecordEnum.ACCOUNT) {
-      // For an account let's assume that the first email is the one we should be querying against
-      const attendeeEmail = emailArray[0];
-      soql = `SELECT Id, Email, OwnerId, AccountId FROM Contact WHERE Email = '${attendeeEmail}' AND AccountId != null`;
-
-      // If this is for a round robin skip then we need to return the account record
-      if (forRoundRobinSkip) {
-        const results = await conn.query(soql);
-        if (results.records.length) {
-          const contact = results.records[0] as { AccountId: string };
-          if (contact) {
-            soql = `SELECT Id, OwnerId FROM Account WHERE Id = '${contact.AccountId}'`;
-          }
-        } else {
-          // If we can't find the exact contact, then we need to search for an account where the contacts share the same email domain
-          const accountId = await this.getAccountIdBasedOnEmailDomainOfContacts(attendeeEmail);
-          if (accountId) {
-            soql = `SELECT Id, OwnerId FROM Account WHERE Id = '${accountId}'`;
+    const log = logger.getSubLogger({ prefix: [`[getContacts]:${emails}`] });
+    try {
+      const conn = await this.conn;
+      const emailArray = Array.isArray(emails) ? emails : [emails];
+
+      const appOptions = this.getAppOptions();
+      const recordToSearch =
+        (forRoundRobinSkip ? appOptions?.roundRobinSkipCheckRecordOn : appOptions?.createEventOn) ??
+        SalesforceRecordEnum.CONTACT;
+
+      let soql: string;
+      let accountOwnerId = "";
+
+      log.info("Getting contacts for emails", {
+        emailArray,
+        includeOwner,
+        forRoundRobinSkip,
+        recordToSearch,
+        appOptions,
+      });
+
+      // Handle Account record type
+      if (recordToSearch === SalesforceRecordEnum.ACCOUNT) {
+        // For an account let's assume that the first email is the one we should be querying against
+        const attendeeEmail = emailArray[0];
+        log.info("Searching account for email", { attendeeEmail });
+
+        soql = `SELECT Id, Email, OwnerId, AccountId FROM Contact WHERE Email = '${attendeeEmail}' AND AccountId != null`;
+
+        // If this is for a round robin skip then we need to return the account record
+        if (forRoundRobinSkip) {
+          const results = await conn.query(soql);
```

Comment:
Followup:
It is best to have type assertion here that considers `records` as possibly nullish and records. Items of records should be asserted as nullish.
In this way if we accidentally miss an optional chaining operator, TS will complain.

Right now records is `unknown[]`, so TS won't complain if we miss a ? after records.